File: strava_api.js
const auth_link = "https://www.strava.com/oauth/token"


function plotActivities(dates, distances){
    data = {dates, distances}
    Plotly.newPlot('strava data', [data], {margin: { t: 0 }} )
}


function getActivities(response){

    const activities_link = `https://www.strava.com/api/v3/athlete/activities?access_token=${response.access_token}`

    fetch(activities_link)
        .then((res) => res.json()) // Parse the response into a JSON
        .then(function(data){
            let dates = [];
            let distances = [];
            for(var x=0; x<data.length; x++){
                let distance = data[x].distance
                let type = data[x].type

                if (type == 'Run'){
                    distance *= 0.9322
                }

                dates.push(data[x].start_date);
                distances.push(distance);
            }
            plotActivities(dates, distances);
        })
}

// If current token is invalid, fetches new token via refresh token
function reAuthorize(){
    fetch(auth_link,{
        method: 'post',
        headers: {
            'Accept': 'application/json, text/plain, */*',
            'Content-Type': 'application/json'
        },

        body: JSON.stringify({

            client_id: '128691',
            client_secret: '5b9731c4e1b1313cc7d5bbaa65371a93a7ec0a86',
            refresh_token: '6b76ea9d526d179ffe5384f88d0396860723eb03',
            grant_type: 'refresh_token'
        })
    })
    .then(res => res.json())
        .then(res => getActivities(res))
}


reAuthorize()

File: strava_api.py
import requests
import urllib3
urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)

auth_url = "https://www.strava.com/oauth/token"
activites_url = "https://www.strava.com/api/v3/athlete/activities"

payload = {
    'client_id': "128691",
    'client_secret': '5b9731c4e1b1313cc7d5bbaa65371a93a7ec0a86',
    'refresh_token': '6b76ea9d526d179ffe5384f88d0396860723eb03',
    'grant_type': "refresh_token",
    'f': 'json'
}

print("Requesting Token...\n")
res = requests.post(auth_url, data=payload, verify=False)
access_token = res.json()['access_token']
print("Access Token = {}\n".format(access_token))

header = {'Authorization': 'Bearer ' + access_token}
param = {'per_page': 200, 'page': 1}
my_dataset = requests.get(activites_url, headers=header, params=param).json()

print(my_dataset[0]["name"])
print(my_dataset[0]["map"]["summary_polyline"])